controllers/confidential_handler.go

```diff
@@ -62,9 +62,11 @@ func (r *KataConfigOpenShiftReconciler) toggleFeatureConfidential(enabled bool)
 	// Changes to make before pods start:
 	//   - Edit peer pods config to enable confidential.
 
-	// Patch peer pods configMap, if it exists.
-	if err := patchConfigMap(r.Client, peerpodsCMName, OperatorNamespace, peerpodsCMData); err != nil && !k8serrors.IsNotFound(err) {
-		return err
+	if r.kataConfig.Spec.EnablePeerPods {
+		// Patch peer pods configMap, if it exists.
+		if err := patchConfigMap(r.Client, peerpodsCMName, OperatorNamespace, peerpodsCMData); err != nil && !k8serrors.IsNotFound(err) {
+			return err
+		}
```

Comment:
nit: this commit should be squash with the previous one adding the confidential_handler